content/doc/developer/plugin-development/incrementals.adoc

```diff
@@ -0,0 +1,108 @@
+---
+layout: developersection
+title: "Developing plugins and Jenkins core in parallel"
+tags:
+- jenkins
+- incrementals
+- core
+- plugin
+description: "This article explains how to use Incrementals for developing plugins in parallel with Jenkins core"
+---
+
+A common problem one might stumble upon is developing an API in Jenkins core and simultaneously
+building a reference implementation inside a plugin.
+What to do, one might ask.
+
+Worry not, Incrementals are here to the rescue!
+
+With incrementals, we can create Jenkins core incremental versions which can be used along with the plugin to build the
+plugin against the very latest changes you have made to core.
```

Comment:
This is only scenario. Equally, perhaps more, important in practice is developing an API in a base plugin (an “upstream” PR) and consuming it in a more specific plugin (a “downstream” PR).

Reply:
The example might be a new addition to an API plugin and a simultaneous change in a consuming plugin that uses the change in the API plugin.  For example, the GitHub plugin might need an addition in the GitHub API plugin or the Git plugin might need an addition in the Git Client plugin.